OrbitBase/include/OrbitBase/Action.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORIBIT_BASE_ACTION_H_
+#define ORIBIT_BASE_ACTION_H_
+
+// Actions are executed by MainThreadExecutor
+// The Action is an abstract class which can
+// be executed. See below for basic implementations for
+// this class: VoidAction and UnaryAction.
+class Action {
+ public:
+  Action() = default;
+  virtual ~Action() = default;
+
+  // Executes the action.
+  virtual void Execute() = 0;
+};
+
+// This class implements an action without parameters.
+template <typename F>
+class LambdaAction : public Action {
```

Comment:
Why is this called lambda action? Is this a standard name? "Lambda" implies that arguments can be specified and thus an anonymous function can be defined. This is not the case here and seems to be misleading.

Reply:
I am open to suggestions. FunctorAction has the same problem, ActionWithFunctorWithoutArguments seems too specific given that there is capture that can be used to pass arguments to lambda expression.

Reply:
renamed to NullaryFunctorAction for now.